### DISCUSSION TITLE
which version of yolo is free for commercial use if any thanks..

### DISCUSSION THREAD

**Discussion by master2142** [ultralytics/ultralytics#3974] [General]
and also is this just becoming a paid service like redhat and Rasa  and java thanks.

**Comment by jelena2712** [comment]
As I understood, it still can be used for commercial purposes, but I am not sure I understand the conditions?
Can someone please give a brief explanation under which conditions yolov5 can be used in commercial products?
Thanks!

**Reply by glenn-jocher** [reply]
@jelena2712 hello! YOLOv8, like other Ultralytics YOLO models, is available under two licenses: the AGPL-3.0 License and the Enterprise License. The AGPL-3.0 License allows for commercial use, but it requires that any modifications to the code or derivative works also be open-sourced under the same license. If you're looking to integrate YOLOv8 into a commercial product without open-sourcing your proprietary code, the Enterprise License would be the appropriate choice, providing more flexibility for commercial product development.

It's important to note that while we do offer these licensing options, Ultralytics is committed to supporting the open-source community. The AGPL-3.0 License ensures that improvements are shared, benefiting everyone, while the Enterprise License offers a path for commercial applications that need to keep their codebase private.

For detailed information on the conditions and to ensure compliance with the licensing terms, please refer to the licensing section in our documentation at https://docs.ultralytics.com. 😊

**Reply by thecoder00007** [reply]
Hi @glenn-jocher,

today, I tried to find any information about this new license, (I already mentioned you in other discussion).
1- We didn't see any information about license price.
2- In the case I used Yolov5 or 8 to train a custom model, and then I used this model to do inference in my company (for internal purpose not to sell it). In this case the code for training and inference existing already. a) I don't know which kind on sharing we need to do? b) where and how, should share the code, if it's worth b) If we trained more than custom models (two diff. case) and the process (even the code) are the same?

Thank you

**Reply by glenn-jocher** [reply]
Hi there! 👋

Great questions! Let's clarify:

1. **License Price:** For the Enterprise License pricing, it's best to contact us directly at contact @thecoder00007.com. We'll provide you with all the necessary details tailored to your specific use case.

2. **Using YOLOv5/YOLOv8 for Internal Purposes:**
   - **a) Sharing Requirements:** If you're using the AGPL-3.0 licensed version of YOLOv5/YOLOv8 and you modify the source code or use it in a way that creates derivative works, you're required to share those modifications under the same license. 
   - **b) How and Where to Share:** Should you modify the source code or create derivative works, you should make your modified source code available to those who interact with your modified version, typically by hosting it on a platform like GitHub.
   - **c) Multiple Custom Models:** If the process and code for training/inference on different models remain unchanged, you're essentially in the same scenario as described above. The key factor is whether you modify the YOLO source code or create derivative works based on it.

If your use case requires keeping your modifications private or you seek more flexibility, the Enterprise License might be the right choice for you. Feel free to reach out for more details on licensing options that best suit your needs!

Hope this helps! Let us know if you have any more questions. 😊

**Reply by thecoder00007** [reply]
@glenn-jocher,
Thank you for your answer. I need more clarification for this answer

**_> a) Sharing Requirements: If you're using the AGPL-3.0 licensed version of YOLOv5/YOLOv8 and you modify the source code or use it in a way that creates derivative works, you're required to share those modifications under the same license. However, if you're just using the model for inference without modifying the source code, there's no requirement to share your internal code._**

if we train a custom model with our dataset, then use this model to do inference. Do we need to share the code?
thanks again

**Reply by Burhan-Q** [reply]
@thecoder00007 you be required to publicly share and open source the modified code, but also the dataset, models, and any code relevant to using the model as well. This is part of the _derivative works_ mentioned in the AGPL-3.0 terms. The Ultralytics Enterprise license will grant you or your organization permission to keep all works private while maintaining an active license. You can visit our [licensing page](https://ultralytics.com/license) to learn more or request an Enterprise License.

**Reply by thecoder00007** [reply]
@Burhan-Q But in this case I didn't change the code.

**Reply by thecoder00007** [reply]
Also, according to AGPL-3.0 license, the distribution of the solution, and the word derivative it is not mentioned or defined

**Reply by thecoder00007** [reply]
@Burhan-Q  Here is the answer of Glenn in this discussion: https://github.com/ultralytics/yolov5/issues/4716
**_The AGPL-3.0 license allows the use of YOLOv5 for commercial purposes, including projects for freelance clients. You can train the YOLOv5 model using your freelance clients' custom dataset and use it in your projects without needing to pay for an Enterprise license._**

**Reply by Burhan-Q** [reply]
@thecoder00007 I understand your confusion, it's a complex document and the jargon can be rather opaque. If you need legal advice, I _highly_ recommend seeking professional legal counsel, as they will be able to provide you with an in depth analysis of the license and how it translates to your exact circumstance. 

As employee of Ultralytics, I can assure you that the intent of AGPL-3.0 means that _all works_ related or pertaining to the application, implementation, or modification of Ultralytics **must** be publicly made available using the same AGPL-3.0 terms. All distributions of Ultralytics work modified or unmodified, along with works related in anyway to the Ultralytics source code, are required to be made publicly open source via AGPL-3.0 _or_ can be licensed under an Enterprise License Agreement directly from Ultralytics.

**Reply by glenn-jocher** [reply]
@thecoder00007 I've updated the thread you linked to. As Burhan has clarified (thank you Burhan!) AGPL-3.0 requires you to open-source any downstream projects that use Ultralytics models or code, including larger projects that contain a licensed model or code, under the same AGPL-3.0 license. 

We're super happy to provide an alternate Enterprise License if you need to keep your projects closed-source or proprietary. Many of the world's largest companies are using our models and code under Enterprise Licenses, they've proven very popular :) 

If you'd like to sync up with our licensing team please fill out the form at https://ultralytics.com/license

**Reply by nullbyte91** [reply]
@Burhan-Q @glenn-jocher Just to confirm, if we don't modify the code and use the open-source data for transfer learning, and perform inference in a production environment, then we are compliant with the AGPL license, correct?

**Reply by Burhan-Q** [reply]
@nullbyte91 incorrect. You **must also** publicly publish **all related** source code, models, and dataset using the same AGPL-3.0 terms. This means whatever software is used for inference, parsing results, etc. is required to be publicly open sourced using AGPL-3.0. You can instead request and purchase an Enterprise License Agreement from https://ultralytics.com/license that will grant use _without_ the AGPL-3.0 open source requirement.
